### ISSUE TITLE
npm refuses to install packages

### ISSUE THREAD

**Issue by Hackzzila** [npm/npm#16950]
#### I'm opening this issue because:

  - [x] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [x] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?

npm 5 refuses to install [co](https://www.npmjs.com/package/co). npm 4 seems to work on node 7. Other packages install just fine.

```
npm ERR! Cannot read property 'find' of undefined
npm ERR! Cannot read property 'find' of undefined
npm ERR! Cannot read property 'find' of undefined

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/hackzzila/.npm/_logs/2017-06-03T17_06_10_856Z-debug.log
```

[npm-debug.log](https://gist.github.com/Hackzzila/c2035c4aac8c1ca19c607b6162ad5466)

#### How can the CLI team reproduce the problem?

`npm i co`

### supporting information:

 - `npm -v` prints: `5.0.2`
 - `node -v` prints: `v8.0.0`
 - `npm config get registry` prints: `https://registry.npmjs.org/`
 - Windows, OS X/macOS, or Linux?: Ubuntu 16.04 with Bash on Ubuntu on Windows
 - Network issues:
   - Geographic location where npm was run: Minnesota, USA
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by Hackzzila** [comment]
Huh, I tried one more time (after trying 4-5 times) and it seems to have fixed itself.

**Comment by Hackzzila** [comment]
I tried to run `npm link` and I got the same error.

**Comment by pszabop** [comment]
same here on docker node:latest, npm 5.0, on a mounted volume.  Non-mounted volume appears to work.

  I wish they'd snapshotted a docker image on 7.x...  good grief.  I've been down twice now in 3 days.

**Comment by biscoe916** [comment]
I'm 100% having issues where NPM 5 is selectively not installing certain modules. I can't figure out how to consistently reproduce the problem. **Usually**, if I `rm -rf node_modules` it'll correct itself, but not always.

**Comment by pszabop** [comment]
Here's a workaround:  `rm package-lock.json`

That completely fixed the problem for me.

**Comment by biscoe916** [comment]
Yep, that worked for me too. Thanks!

**Comment by pszabop** [comment]
I note that the lock file keeps hanging around after a successful install (5.0)...I think there's an issue for that already.  next time you run npm, I'm guessing you have to remove it again.

**Comment by zkat** [comment]
This should be fixed by `5.0.3`.
